### PR TITLE
Logging service

### PR DESCRIPTION
**Pull request by DenverM80**
Create a LoggingService to inject into presenters in order to reduce coupling.  Update all references to (ds3Common.)deepStorageBrowserPresenter.logText() to use LoggingService.logMessage()

### CODE REVIEW

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java [hunk 1]
```diff
@@ -313,7 +318,7 @@ else if (Guard.isNullOrEmpty(values)) {
 
         } catch (final Exception e) {
             LOG.error("Failed to transfer data to black pearl", e);
-            ds3Common.getDeepStorageBrowserPresenter().logText("Failed to transfer data to black pearl: " + e, LogType.ERROR);
+            loggingService.logMessage("Failed to transfer data to black pearl: " + e, LogType.ERROR);
```
Comment (berkeleysquare) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java]: Haven't seen enough to know what % is out of resource or internationalizable, but is it worth forcing all error messages?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/logservice/ApplicationLoggerSettings.java [hunk 2]
```diff
@@ -46,7 +61,7 @@ public PatternLayoutEncoder updateLogBackSettings(final String pattern) {
         fileAppender.setFile(destPath.toString());
 
         final PatternLayoutEncoder layout = new PatternLayoutEncoder();
-        layout.setPattern(pattern);
+        layout.setPattern("%date %level [%thread] %logger{10} [%file:%line] %msg%n");
```
Comment (berkeleysquare) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/logservice/ApplicationLoggerSettings.java]: Static string for pattern?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/logservice/ApplicationLoggerSettings.java [hunk 3]
```diff
@@ -83,10 +98,12 @@ public PatternLayoutEncoder updateLogBackSettings(final String pattern) {
         if (logSettings.getDebugLogging()) {
             rootLogger.setLevel(Level.DEBUG);
         } else {
-            rootLogger.setLevel(Level.ERROR);
+            rootLogger.setLevel(Level.INFO);
```
Comment (berkeleysquare) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/logservice/ApplicationLoggerSettings.java]: Performance hit on wider root logger? Does the default need to be changed?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/Main.java [hunk 4]
```diff
@@ -35,17 +37,20 @@ public void start(final Stage primaryStage) throws Exception {
...
         final SavedJobPrioritiesStore savedJobPrioritiesStore = SavedJobPrioritiesStore.loadSavedJobPriorties();
         final JobInterruptionStore jobInterruptionStore = JobInterruptionStore.loadJobIds();
         final SavedSessionStore savedSessionStore = SavedSessionStore.loadSavedSessionStore();
         final ResourceBundle resourceBundle = ResourceBundle.getBundle("lang", new Locale(ConfigProperties.getInstance().getLanguage()));
         final DataFormat dataFormat = new DataFormat(resourceBundle.getString("dataFormat"));
         final JobWorkers jobWorkers = new JobWorkers(settings.getProcessSettings().getMaximumNumberOfParallelThreads());
         final Logger injectorLogger = LoggerFactory.getLogger("Injector");
-        LOG.info("Starting Deep Storage Browser v2.0");
+        final LoggingServiceImpl loggingService = new LoggingServiceImpl();
+        LOG.info("Starting Deep Storage Browser v2.1");
```
Comment (DenverM80) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/Main.java]: I actually have that ready to go in a separate PR
https://github.com/SpectraLogic/ds3_java_browser/pull/32

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java [hunk 5]
```diff
@@ -111,11 +112,22 @@
...
 
+    @Inject
+    private LoggingService loggingService;
+
     private DeepStorageBrowserTaskProgressView<Ds3JobTask> jobProgressView;
 
     @Override
     public void initialize(final URL location, final ResourceBundle resources) {
         try {
+            loggingService.registerLogListener( (message, logType) -> {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java]: This should be pulled out into it's own method, and then use a method reference here to register it.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/createbucket/CreateBucketPresenter.java [hunk 6]
```diff
@@ -99,13 +103,15 @@ public void createBucket() {
                         createBucketWithDataPoliciesModel.getSession().getClient(), bucketNameField.getText().trim(),
                         deepStorageBrowserPresenter);
                 workers.execute(createBucketTask);
-                createBucketTask.setOnSucceeded(event -> Platform.runLater(() -> {
+                createBucketTask.setOnSucceeded(event -> {
                     LOG.info("Bucket is created");
                     deepStorageBrowserPresenter.logText(resourceBundle.getString("bucketCreated"), LogType.SUCCESS);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/createbucket/CreateBucketPresenter.java]: You missed a place where you should be using the loggerservice

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTableItem.java [hunk 7]
```diff
@@ -27,10 +24,14 @@
     private final Ds3Common ds3Common;
     private boolean accessedChildren = false;
     private TreeTableView ds3TreeTable;
-    private DeepStorageBrowserPresenter deepStorageBrowserPresenter;
-
-    public Ds3TreeTableItem(final String bucket, final Session session, final Ds3TreeTableValue value, final Workers
-            workers, final Ds3Common ds3Common) {
+    private LoggingService loggingService;
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTableItem.java]: This looks like it should be made final.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java (hunk 1 above)
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java]: Ideally, we shouldn't be logging exceptions into the gui logger.  If we can we should map it to a user friendly message.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/logservice/ApplicationLoggerSettings.java (hunk 3 above)
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/logservice/ApplicationLoggerSettings.java]: This is probably not something that we need to worry about with INFO.  It will help us with debugging in the field if it is set to this.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java [hunk 8]
```diff
@@ -111,11 +112,22 @@
     @Inject
     private Workers workers;
 
+    @Inject
```
Comment (DenverM80) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java]: Definitely going to go back and do a more pointed PR for the GuicePresenterInjector and GuiModule either next or soon

**Comment by DenverM80** [comment]
updated.

**Comment by DenverM80** [comment]
updated  LoggingService with JavaRx SubjectPublisher

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/SearchJobTask.java [hunk 9]
```diff
@@ -63,24 +71,12 @@ public SearchJobTask(final List<Bucket> searchableBuckets, final String searchTe
             return list;
         } catch (final Exception e) {
             LOG.error("Search failed", e);
-            printLog(StringBuilderUtil.searchFailedMessage().toString() + e, LogType.ERROR);
+            loggingService.logMessage(StringBuilderUtil.searchFailedMessage().toString() + e, LogType.ERROR);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/SearchJobTask.java]: Let's clean this up a bit.  The `searchFailedMessage()` call returns a StringBuilder, which we immediately turn into a String and then concatenate the exception's `toString()` method to it.  Instead, after we get the StringBuilder from `searchFailedMessage()`, append the exception to it, then call `toString()`
